Automattic/wp-calypso
# Themes - add CTA for community themes on themes page.

Related to issue # https://github.com/Automattic/wp-calypso/issues/94495

related to this PR https://github.com/Automattic/wp-calypso/pull/94264 which removed the upsell banner that previously held the CTA.

## Proposed Changes

Adds CTA to community themes
* Adds a `UpgradePlanForDotorgThemes` theme option.
    * We use this when the theme tier is community and the site does not have theme installation capability.
* Removes the forced fallback to the "activate" label for when we are logged in and have a site selected. Labels are defined in theme options and these were always being bypassed due to this. Where we set the default options, we have many cases within loggedIn && siteId logic that all seem overridden by this return.

BEFORE ( site level )

<img width="600" alt="Screenshot 2024-09-12 at 2 28 31 PM" src="https://github.com/user-attachments/assets/110b4a04-bb9a-447b-97a9-e50c9f1eafe2">

AFTER ( site level )
<img width="600" alt="Screenshot 2024-09-12 at 2 28 17 PM" src="https://github.com/user-attachments/assets/4b6737c1-41fd-4d55-95cd-98b09bd84de3">

BEFORE (global)
<img width="600" alt="Screenshot 2024-09-12 at 2 30 00 PM" src="https://github.com/user-attachments/assets/1007d04a-1d04-4ce2-b8a9-50646ab82250">

AFTER (global)
<img width="600" alt="Screenshot 2024-09-12 at 2 29 49 PM" src="https://github.com/user-attachments/assets/d65e5f5f-7264-4834-87b1-9fdf4ab361b9">

BEFORE/AFTER (LOGGED OUT) remains unaffected:
<img width="600" alt="Screenshot 2024-09-12 at 2 31 45 PM" src="https://github.com/user-attachments/assets/faf0fe8e-9eec-4295-ba5a-0315dd6c8b1a">


## Why are these changes being made?


* Community themes pages feel broken with no CTA.

## Testing Instructions



Community theme for low plan site
* View a community theme in the showcase logged in with a simple/free site.
* Verify there is a CTA to upgrade the plan.
* Go through the upgrade flow from here and verify it works as expected.

Community theme with plan
* View a community theme on a site with a business plan.
* Verify the CTA prompts activation and activates the theme.

Community theme in global context
* View a community theme on the global showcase (/themes with no site slug, parallel to /sites).
* Verify there is a CTA, and that choosing a site leads to the individual theme page.

* Test other theme types and plans, verify the CTA labels are as expected.

## Pre-merge Checklist
